tests/Integration/PrestaShopBundle/Test/WebTestCase.php

```diff
@@ -138,6 +171,92 @@ protected function enableDemoMode()
         $configurationMock->method('get')
             ->will($this->returnValueMap($values));
 
-        self::$kernel->getContainer()->set('prestashop.adapter.legacy.configuration', $configurationMock);
+        $this->container->set('prestashop.adapter.legacy.configuration', $configurationMock);
+    }
+
+    /**
+     * Little helper to retrieve service from Container.
+     *
+     * @param $serviceName
+     *
+     * @return \StdClass
+     * @throws \Exception
+     */
+    protected function get($serviceName)
+    {
+        return $this->container->get($serviceName);
+    }
+
+    /**
+     * Retrieves an instance of Module by module name
+     * @param $moduleName
+     *
+     * @return Module
+     * @throws \Exception
+     */
+    protected function getModule($moduleName)
+    {
+        return $this->get('prestashop.core.admin.module.repository')->getModule($moduleName);
+    }
+
+    /**
+     * Copy a module from "module" folder in "tests" folder to the right one.
+     * Also install the module.
+     *
+     * @param string $moduleName
+     * @return bool
+     * @throws \Exception
+     */
+    protected function installModule($moduleName)
+    {
+        if (ModuleHelper::addModule($moduleName)) {
+            $this->getModule($moduleName)->onInstall();
+            $this->clearCache();
```

Comment:
finally not a good idea to clear cache :/

Reply:
the question is why do you want to clear cache right after installation of module?

Reply:
@mickaelandrieu cache of the entire app should be cleared at the start of every test suite right?

Reply:
The problem I meet currently is that the container of Symfony is not in sync with the list of activated modules.

As we store this list using `kernel.active_modules` parameter in the container and as we rely on this parameter everywhere we want to loop on modules, I can't get my template overriding working in my tests.

So I tried to clear the cache to force the container to be refreshed but it's not working because we can't make a hot-reload of the cache.

Next step, try to force a refresh of the container using a resource checker from Config Cache: it's very uncommon needs in Symfony... Also, we have like a LOT of weird behaviors, sometimes our mocks don't work as expected or are missing or whatever... we are spending a lot of times on it because we want 1) test all our natives modules with ONE and one reliant way to do it 2) we want to help module contributors to test their own modules.

This sounds easy, but It's really hard :/

Reply:
@mickaelandrieu yeah it sounds easy, like "ok, we have new modules, lets reload application" but that would be bad for tests time i guess

side question: from what i understand you guys are planning to refacto native modules like "ps_imageslider" etc. to make it Symfony way too, right?

Reply:
@kpodemski I'm not sure of the scope of what we will do on modules for now, but our main objective is to make the modules will we redo "really good modules" so the community can use/adapt them to make good modules too. Like docs, but better 👍

Reply:
More as we said together yesterday, clearCache only clear 'dev' or 'prod' cache :)